File: icamera/src/main/java/me/shouheng/icamera/enums/CameraFace.java
package me.shouheng.icamera.enums;

import android.support.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static me.shouheng.icamera.enums.CameraFace.*;

/**
 * Camera face
 *
 * @author <a href="mailto:shouheng2015@gmail.com">WngShhng</a>
 * @version 2019-12-28 16:38
 */
@IntDef(value = {FACE_REAR, FACE_FRONT})
@Retention(value = RetentionPolicy.SOURCE)
public @interface CameraFace {

    /** Rear camera */
    int FACE_REAR           = 0x0000;

    /** Front camera */
    int FACE_FRONT          = 0x0001;
}

File: icamera/src/main/java/me/shouheng/icamera/enums/CameraFace.kt
package me.shouheng.icamera.enums

import android.support.annotation.IntDef

/**
 * Camera face
 *
 * @author [WngShhng](mailto:shouheng2015@gmail.com)
 * @version 2019-12-28 16:38
 */
@IntDef(value = [CameraFace.FACE_REAR, CameraFace.FACE_FRONT])
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class CameraFace {
    companion object {
        /** Rear camera  */
        const val FACE_REAR = 0x0000
        /** Front camera  */
        const val FACE_FRONT = 0x0001
    }
}